bindings/encoding/input_test.go

```diff
@@ -84,6 +85,50 @@ func TestEncodeProveBlockInvalidInput(t *testing.T) {
 	require.NotNil(t, encoded)
 }
 
+func TestEncodeBlockMetadata(t *testing.T) {
+	// since strings are right padded in solidity https://github.com/ethereum/solidity/issues/1340
+	var (
+		arr  = common.RightPadBytes([]byte("abcd"), 32)
+		abcd [32]byte
+	)
+	copy(abcd[:], arr)
+
+	// Encode block metadata using EncodeBlockMetadata function
+	encoded, err := EncodeBlockMetadata(&bindings.TaikoDataBlockMetadata{
+		Id:                uint64(1),
+		L1Height:          uint64(1),
+		L1Hash:            abcd,
+		Beneficiary:       common.HexToAddress("0x10020FCb72e27650651B05eD2CEcA493bC807Ba4"),
+		Treasury:          common.HexToAddress("0x50081b12838240B1bA02b3177153Bca678a86078"),
+		TxListHash:        abcd,
+		TxListByteStart:   big.NewInt(0),
+		TxListByteEnd:     big.NewInt(1000),
+		GasLimit:          1,
+		MixHash:           abcd,
+		Timestamp:         uint64(1),
+		DepositsProcessed: []bindings.TaikoDataEthDeposit{},
```

Comment:
Seems we only test the `metaData` with an empty `DepositsProcessed` list? Is it necessary to also test it with another one with non-empty `DepositsProcessed` list?